token/services/db/sql/common/query.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package common
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	SelectStatement         = `SELECT`
+	SelectDistinctStatement = `SELECT DISTINCT`
+	InsertStatement         = `INSERT INTO`
+	UpdateStatement         = `UPDATE`
+)
+
+type Select struct {
+	stmt    string
+	columns []string
+	from    []string
+	where   string
+	orderBy string
+}
+
+func NewSelect(columns ...string) *Select {
+	return &Select{
+		stmt:    SelectStatement,
+		columns: columns,
+	}
+}
+
+func NewSelectDistinct(columns ...string) *Select {
+	return &Select{
+		stmt:    SelectDistinctStatement,
+		columns: columns,
+	}
+}
+
+func (s *Select) From(tables ...string) *Select {
+	s.from = tables
+	return s
+}
+
+func (s *Select) Where(where string) *Select {
+	s.where = where
+	return s
+}
+
+func (s *Select) Compile() (string, error) {
```

Comment:
I would leave compile to an interpreter. Different SQL implementations have different results (e.g. check the common Interpreter and Postgres Interpreter). So each db implementation could be injected its own interpreter like is already done for the conditions.
```
type Interpreter interface {
 Select(s Select) (string, error)
 Insert(I Insert) (string, error)
}
```